x-pack/legacy/plugins/ml/common/types/custom_urls.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { AnomalyRecordDoc } from './anomalies';
+
+export interface UrlConfig {
+  url_name: string;
+  url_value: string;
+}
+
+export interface KibanaUrlConfig extends UrlConfig {
+  time_range: string;
+}
+
+export interface AnomalyRecordSource extends AnomalyRecordDoc {
```

Comment:
I'd give this interface a name which links it clearly to custom urls. `CustomUrlAnomalyRecordDoc` ?